client/pages/Event/EventDetailsPage.jsx

```diff
@@ -41,7 +42,10 @@ class EventDetailsPage extends React.Component {
   }
 
   static getDerivedStateFromProps(nextProps, prevState) {
-    if (nextProps.events.status && (nextProps.events !== prevState.events && prevState.updated === false)) {
+    if (
+      nextProps.events.status
+      && (nextProps.events !== prevState.events && prevState.updated === false)
```

Comment:
Nice implementation @Alameen688 but I am curious to know why a prop is getting stored in the state.

Reply:
@phemmelliot I didn't implement this I only format the code that's why it appeared as a change. However, based on my understanding of the implementation, the "updated" state attribute is used to determine if the component should re-render. An example use-case here is when the edit event modal pops up and the user changes the event detail, once the user saves the edit and it was successful the modal closes and the component re-renders to reflect the update.

You could try this out by removing the getDerivedStateFromProps and try to update an event you created, you would notice that the changes will not reflect until you reload but if the hook is there it will update instantly has the component would know that the UI should be re-rendered as the "updated" state has changed